packages/agent/src/agent/http/index.ts

```diff
@@ -120,6 +121,11 @@ export interface HttpAgentOptions {
    * @default 3
    */
   retryTimes?: number;
+  /**
+   * Whether the agent should verify signatures signed by node keys on query responses. Increases security, but adds overhead and must make a separate request to cache the node keys for the canister's subnet.
+   * @default true
+   */
+  verifyQuerySignatures?: boolean;
```

Comment:
It seems setting verifyQuerySignatures to false only skip verifying signature but not fetching node keys, right?

Reply:
I cannot comment on line 118 about `disableNonce` so I put it here. 
`disableNonce` is still false by default. When and how we make it true by default for only queries?

Reply:
This is resolved in #792

Reply:
Just to clarify, #792 only solves the second comment but not the first comment, right?

> It seems setting verifyQuerySignatures to false only skip verifying signature but not fetching node keys, right?

Reply:
Nvm, seems that fetching nodes keys is only done when verifyQuerySignatures is false. This is checked on line 511.